terraform/layers/app/alb.tf

```diff
@@ -0,0 +1,87 @@
+
+data "aws_vpc" "grad_lab_1_vpc" {
+  filter {
+    name   = "tag:Name"
+    values = ["${local.resource_name_prefix}-vpc"]
+  }
+}
+
+data "aws_subnets" "grad_lab_1_vpc_public_subnets" {
+  filter {
+    name   = "vpc-id"
+    values = [data.aws_vpc.grad_lab_1_vpc.id]
+  }
+
+  filter {
+    name   = "tag:Name"
+    values = ["${local.resource_name_prefix}-vpc-public-*"]
+  }
+}
+
+module "grad_lab_1_alb" {
+  source = "terraform-aws-modules/alb/aws"
+
```

Comment:
I think I've missed this on modules used for other resources, but we should consider pinning the version to avoid breaking code with major updates. So something like version = "~> 9.12.0". 

The ~> means we will accept any versions that match 9.*, but ignore any major module changes that could cause breakages